src/docs/resources/architectural-overview.md

```diff
@@ -0,0 +1,966 @@
+time:
+
+- Parents can dictate the size of a child object by setting maximum and minimum
+  constraints to the same value. For example, the topmost render object in a
+  phone app constrains its child to be the size of the screen. (Children can
+  choose how to use that space. For example, they might just center what they
+  want to render within the dictated constraints.)
+- A parent can dictate the child’s width but give the child flexibility over
+  height (or dictate height but offer flexible over width). A real-world example
+  is flow text, which might have to fit a horizontal constraint but vary
+  vertically depending on the quantity of text.
+
+This model works even when a child object needs to know how much space it has
+available to decide how it will render its content. By using a
+[`LayoutBuilder`](https://api.flutter.dev/flutter/widgets/LayoutBuilder-class.html)
+widget, the child object can examine the passed-down constraints and use those
+to determine how it will use them, for example:
+
+```dart
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      builder: (context, constraints) {
+        if (constraints.maxWidth < 600) {
+          return OneColumnLayout();
+        } else {
+          return TwoColumnLayout();
+        }
+      },
+    );
+```
+
+More information about the constraint and layout system, along with worked
+examples, can be found in the [Understanding
+Constraints](https://flutter.dev/docs/development/ui/layout/constraints) topic.
+
+The root of all `RenderObject`s is the `RenderView`, which represents the total
+output of the render tree. When the platform demands a new frame to be rendered
+(for example, because of a
+[vsync](https://source.android.com/devices/graphics/implement-vsync) or because
+a texture decompression/upload is complete), a call is made to the
+`compositeFrame()` method, which is part of the `RenderView` object at the root
+of the render tree. This creates a `SceneBuilder` to trigger an update of the
+scene. When the scene is complete, the `RenderView` object passes the composited
+scene to the `Window.render()` method in `dart:ui`, which passes control to the
+GPU to render it.
+
+Further details of the composition and rasterization stages of the pipeline are
+beyond the scope of this high-level article, but more information can be found
+[in this talk on the Flutter rendering
+pipeline](https://www.youtube.com/watch?v=UUfXWzp0-DU).
+
+## Platform embedding
+
+As we’ve seen, rather than being translated into the equivalent OS widgets,
+Flutter user interfaces are built, laid out, composited, and painted by Flutter
+itself. The mechanism for obtaining the texture and participating in the app
+lifecycle of the underlying operating system inevitably varies depending on the
+unique concerns of that platform. The engine is platform-agnostic, presenting a
+[stable
+ABI](https://github.com/flutter/engine/blob/master/shell/platform/embedder/embedder.h)
```

Comment:
I had to look up 'ABI'. Application Binary Interface?

Reply:
Yes. I've expanded it.